### PR TITLE
feat: add feature flag for avax

### PR DESCRIPTION
**Pull request by cbachmeier**
## Description

- Add feature flag to gate Avalanche launch

_Slack thread:_ https://uniswapteam.slack.com/archives/C052ARNFZGU/p1689010169325719

## Screen capture
avaxGate

### CODE REVIEW

**Comment by zzmp** [comment]
FYI merging this will immediately deploy to prod

File: src/featureFlags/flags/avalanche.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  return useBaseFlag(FeatureFlag.avalanche)
+}
+
+export function useIsAvalancheEnabled(): boolean {
+  return useAvalancheFlag() === BaseVariant.Enabled
+}
+
+export function filterChainsForAvalanche(chains: ChainId[]): ChainId[] {
+  return chains.filter((chain) => chain !== ChainId.AVALANCHE || useIsAvalancheEnabled())
+}
```
Comment (cartcrom) [inline src/featureFlags/flags/avalanche.ts:15]: I'm confused on how this function is able to use a hook... is useIsAvalancheEnabled not actually a react hook?

wouldn't this need to be something like
```suggestion
export function useFilterChainsForAvalanche(chains: ChainId[]): ChainId[] {
  const isAvalancheEnabled = useIsAvalancheEnabled()
  return chains.filter((chain) => chain !== ChainId.AVALANCHE || isAvalancheEnabled)
}
```

but clearly it's compiling so I must be missing something here?

Comment (cbachmeier) [reply src/featureFlags/flags/avalanche.ts]: I've updated the fn to follow hook naming conventions

**Review by cartcrom — APPROVED** [review]

File: src/components/NavBar/SearchBarDropdown.tsx [hunk 2]
```diff
@@ -356,8 +357,8 @@ export const SearchBarDropdown = ({
     trendingCollectionsAreLoading,
     shouldDisableNFTRoutes,
   ])
-
-  const showChainComingSoonBadge = chainId && BACKEND_NOT_YET_SUPPORTED_CHAIN_IDS.includes(chainId) && !isLoading
+  const gatedUnsupportedChains = useFilterChainsForAvalanche([...BACKEND_NOT_YET_SUPPORTED_CHAIN_IDS])
```
Comment (vm) [inline src/components/NavBar/SearchBarDropdown.tsx:360]: nit: don't think you need to spread this?

**Review by vm — APPROVED** [review]

**Review by JFrankfurt — APPROVED** [review]

**Comment by cbachmeier** [comment]
For anyone using this as a future reference for adding a new chain feature flag, the gate logic in `src/theme/components/RadialGradientByChainUpdater.ts` to turn on the background gradient worked locally but did not work in prod. A new solution will need to be determined.

**Pull request merged**